main.rb

```diff
@@ -0,0 +1,117 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    counter = 0
+    while counter < size
+      if instance_of(Hash) || instance_of?(Range)
+        new_hash = to_a
+        yield(new_hash[counter])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+  end
+
+  def my_each_with_index
+    counter = 0
+    while counter < size
+      if instance_of(Hash) || instance_of?(Range)
+        new_hash = to_a
+        yield(new_hash[counter], counter)
+      else
+        yield(self[counter], counter)
+      end
+      counter += 1
+    end
+  end
+
+  def my_select
+    if instance_of?(Hash)
+      new_hash = {}
+      my_each { |key, value| new_hash[key] = value if yield(key, value) }
+      new_hash
+    else
+      arr = []
+      my_each { |item| arr.push(item) if yield(item) }
+      arr
+    end
+  end
+
+  def my_all?(type = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+    elsif type.nil?
+      my_each { |item| return false if item.nil? }
+    else
+      my_each { |item| return false if item.class != type && item.class.superclass != type }
+    end
+    true
+  end
```

Comment:
#### my_all?
- [ ] 8. #my_all? when no block or argument is given returns false when one of the collection members are false or nil
- [ ] 9. #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] 10. #my_all? when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern